dm/config/subtask.go

```diff
@@ -124,6 +124,26 @@ func (db *DBConfig) Adjust() {
 	AdjustTargetDBTimeZone(db)
 }
 
+// Clone returns a deep copy of DBConfig. This function only fixes data race when adjusting Session.
+func (db *DBConfig) Clone() (DBConfig, error) {
```

Comment:
Shall we just depend on pointer dereference to avoid this kind of special logic and notorious error handling.

Reply:
not understand, pointer dereferencing is  one level shallow copy, at some cases it will cause data race.

Do you mean when we meet data race, we should use lock or adjust the logic to avoid concurrent accessing?

Reply:
I mean we should manually do derefernce for all the pointers. Depend on marshal is some what weird and not ergonomic.

Reply:
BWT, Clone an object may return an error is counter-intuitive